rest_framework/filters.py

```diff
@@ -100,12 +101,18 @@ def filter_queryset(self, request, queryset, view):
 
         orm_lookups = [self.construct_search(six.text_type(search_field))
                        for search_field in search_fields]
-
+        and_queries = []
         for search_term in self.get_search_terms(request):
             or_queries = [models.Q(**{orm_lookup: search_term})
                           for orm_lookup in orm_lookups]
-            queryset = queryset.filter(reduce(operator.or_, or_queries)).distinct()
-
+            and_queries.append(reduce(operator.or_, or_queries))
+
+        if and_queries:
+            if settings.DATABASES[queryset.db]["ENGINE"] == "django.db.backends.oracle":
```

Comment:
We'll want a brief comment here regarding needing this behavior for oracle and distinct.